Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java

```diff
@@ -121,13 +123,27 @@ public void onDirectorySelected(String dir)
 
   public void installWAD(String file)
   {
-    if (NativeLibrary.InstallWAD(file))
+    Thread installWADThread = new Thread(() ->
     {
-      Toast.makeText(mContext, R.string.wad_install_success, Toast.LENGTH_SHORT).show();
-    }
-    else
-    {
-      Toast.makeText(mContext, R.string.wad_install_failure, Toast.LENGTH_SHORT).show();
-    }
+      Looper.prepare();
```

Comment:
I haven't seen this solution of using a Looper on a separate thread in order to do UI stuff directly on that thread. I'm assuming it works if you've tested it, but I can't find any documentation on whether it's actually thread-safe... The more common solution would be to use `runOnUiThread`.